app/Api/Controllers/NotificationController.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace App\Api\Controllers;
+
+use Dingo\Api\Http;
+use Dingo\Api\Routing\Helpers;
+use App\User;
+use App\Notification;
+use App\NotificationAttrib;
+use Illuminate\Http\Request;
+
+class NotificationController extends Controller
+{
+
+    use Helpers;
+
+    public function __construct() {
+    }
+
+    /**
+     * Display a listing of all notifications
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index(Request $request, $type = null)
+    {
+        if ($type === 'archive')
+        {
+            //$notifications = Notification::read([$request->user()->user_id])->limit()->get();
+            $notifications = Notification::select('notifications.*','key')->leftJoin('notifications_attribs', 'notifications.notifications_id', '=', 'notifications_attribs.notifications_id')->where('user_id', $user_id)->where(['key'=>'read', 'value'=> 1]);
```

Comment:
Move these queries into a scope on the Notifications class.
Then the line will look like:
$notifications = Notification::unread();